### PR TITLE
Add build:programs

### PR DESCRIPTION
**Pull request by ripleyaffect**
- Also improves logging

### CODE REVIEW

File: src/commands/build/programs.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    const cwd = process.cwd();
+    const libraryDirectory = path.resolve(cwd, libraryDir);
+    const modulesDirectory = path.resolve(cwd, modulesDir);
+    const programsDirectory = path.resolve(cwd, programsDir);
+
+    this.log(`>> Building programs\n`);
+    this.log(`📍 cwd: ${c.blue(cwd)}`);
+    this.log(`📚 Library: ${c.blue(libraryDirectory)}`);
+    this.log(`📦 Modules: ${c.blue(modulesDirectory)}`);
+    this.log(`🏔  Programs: ${c.blue(programsDirectory)}\n`);
```
Comment (learykara) [inline src/commands/build/programs.js]: xtra space

File: src/tasks/buildPrograms.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  libraryDirectory
+) => {
+  // Get the module
+  const programs = await getYamlFiles(programDirectory);
+
+  log(`Found ${programs.length} program${programs.length === 1 ? '' : 's'}:`);
+  for (const program of programs) {
+    log(`  - ${program.src}`);
+  }
+  log();
```
Comment (learykara) [inline src/tasks/buildPrograms.js]: heh?

File: src/commands/build/programs.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    try {
+      const programs = await buildPrograms(
+        programsDirectory,
+        modulesDirectory,
+        libraryDirectory
+      );
+
+      // Just log for now
+      this.log();
```
Comment (learykara) [inline src/commands/build/programs.js]: heh?

File: src/commands/build/programs.js (hunk 1 above)
Comment (ripleyaffect) [reply src/commands/build/programs.js]: intentional. for some reason mountain is rendered one space less than the others

File: src/commands/build/programs.js (hunk 3 above)
Comment (ripleyaffect) [reply src/commands/build/programs.js]: just adding some spaceing for the logs

**Pull request merged**